merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/utils/JavadocParser.java

```diff
@@ -0,0 +1,175 @@
+package gov.nasa.jpl.aerie.merlin.processor.utils;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class JavadocParser {
+  private static final String TAG_PREFIX = "@aerie";
+  private static final String COMPUTED_ATTRIBUTE_TAG = "%s.computedAttribute".formatted(TAG_PREFIX);
+  // @param is a built-in tag, so we don't prefix it.
+  private static final String PARAM_TAG = "@param";
+  private static final String RESOURCE_TYPE_TAG = "%s.resourceName".formatted(TAG_PREFIX);
+  private static final String UNITS_TAG = "%s.unit".formatted(TAG_PREFIX);
+
+  /**
+   * Parses javadocs searching for resource type + unit paris.
+   * @param elementUtils Utility functions for file parsing.
+   * @param element The element we're parsing the comments for.
+   * @return A map of resource types to their associated unit.
+   */
+  public static Map<String, String> parseMissionModelUnits(Elements elementUtils, TypeElement element) {
+    final var parsedUnits = new HashMap<String, String>();
+
+    // Parse the class javadoc.
+    Optional.ofNullable(elementUtils.getDocComment(element))
+        .map(JavadocParser::removeSingleLeadingSpaceFromEachLine)
+        .map(comment -> {
+          parseComment(comment, parsedUnits, RESOURCE_TYPE_TAG);
+
+          return comment;
+        });
+
+    element
+        .getEnclosedElements()
```

Comment:
From reading this code, it appears that all `@resourceName` annotations are expected to occur in the top-level Mission class. Is this an accurate reading?

I'm wondering whether that would cause resource unit declarations to be written far from the registration of those resources

Reply:
Gotcha - for more complex models, resources tend to be defined in sub-models - e.g:

```java
// Mission.java
public final class Mission {
  public final PowerModel power;
}
```

```java
// PowerModel.java
public final class PowerModel {
  /**
   * @aerie.resourceName /power/soc
   * @aerie.unit percentage
   */
  public final Accumulator batteryStateOfCharge;
}
```

Is this something you feel like we could/should support?

Reply:
I can look into this yes, I think if it's a common use-case we should try and support it!